### DISCUSSION TITLE
Support promises for callbacks like onDelete

### DISCUSSION THREAD

**Discussion by yaswanthmaddula-sureify** [CarlosNZ/json-edit-react#116] [Ideas]
I would like to have confirmation modal, if user tries to delete certain important keys in the json. As the `onDelete` callback is synchronous as of now, the only way for me to achieve this is using window.confirm(). It would be useful if there is a way like promises to support such usecases.

**Comment by CarlosNZ** [comment]
Thanks for the suggestion, I've started an issue to do this: #117
